javalib/src/main/scala/java/io/File.scala

```diff
@@ -541,4 +571,22 @@ object File {
                                minLength = true,
                                throwOnError = true)
 
+  // Ported from Apache Harmony
+  private def checkURI(uri: URI): Unit = {
+    def throwExc(msg: String): Unit =
+      throw new IllegalArgumentException(s"$msg: $uri")
+    def compMsg(comp: String): String = s"Found $comp component in URI"
+
+    if (!uri.isAbsolute) throwExc("URI is not absolute")
+    else if (!uri.getRawSchemeSpecificPart.startsWith("/"))
+      throwExc("URI is not hierarchical")
+    else if (uri.getScheme == null || !(uri.getScheme == "file"))
+      throwExc("Expected file scheme in URI")
+    else if (uri.getRawPath == null || uri.getRawPath.length == 0)
+      throwExc("Expected non-empty path in URI")
+    else if (uri.getRawAuthority != null) throwExc(compMsg("authority"))
+    else if (uri.getRawQuery != null) throwExc(compMsg("query"))
+    else if (uri.getRawFragment != null) throwExc(compMsg("fragment"))
+    // else URI is ok
```

Comment:
Please wrap if/else branch bodies into `{ }` to improve readability.

Reply:
Yes, and done.